test/cask-api-test.el

```diff
@@ -630,6 +630,19 @@
       (cask-install bundle)
       (should (equal orig-load-path load-path)))))
 
+(ert-deftest cask-install-test/pin-archive ()
+  (if  (not (boundp 'package-pinned-packages))
+    (princ "`package-pinned-packages' is not defined, we need Emacs v24.4+ to test the pinning feature.\n")
+    (cask-test/with-bundle
+     '((source localhost)
+       (depends-on "package-a" "0.0.1" :archive "localhost"))
```

Comment:
I'm not sure if we should use a string here if the source is one of the [supported ones](https://github.com/cask/cask/blob/f9e804ea96d14e24ae49eeb72d95d44e358b38f2/cask.el#L159-L166). @lunaryorn What do you think?